boards/silabs/radio_boards/xg23_rb4210a/xg23_rb4210a.dts

```diff
@@ -124,6 +124,7 @@
 	si7021: si7021@40 {
 		compatible = "silabs,si7006";
 		reg = <0x40>;
+		supply-gpios = <&gpioc 9 GPIO_ACTIVE_HIGH>;
```

Comment:
The radios boards have three of these GPIOs: sensor, vcom and display. Introducing `supply-gpios` means we need to patch every driver accordingly.

So, I rather though to use `regulator-fixed` to declare them:

```
       sensor_enable: gpio_switch_0 {
               compatible = "regulator-fixed";
               regulator-name = "sensor_enable";
               regulator-boot-on;
               enable-gpios = <&gpioc 9 GPIO_ACTIVE_HIGH>;
       };
```

Any opinion?

Reply:
I think for now this is the correct solution after reading https://docs.zephyrproject.org/latest/hardware/peripherals/regulators.html. For now, the only user for the regulator will be the Si7006 sensor driver since there isn't a driver for the LCD display yet and the VCOM is a different GPIO.
When the driver for the LCD display is implemented (which I'm considering doing) then we can add the regulator.
BTW, when we do that we will still need to change each driver in order to add the `regualator_enable()` call - Using `regulator-boot-on;` seems like the wrong thing to do since we don't want it on absolutely but rather only when it is needed by a device.
What do you think?

Reply:
I believe we don't need to introduce `regulator-boot-off`. This is already the default behavior if neither `regulator-boot-on` or `regulator-always-on` are present. `drivers/regulator/regulator_common.c:89` seems to properly handle it.

Reply:
For information, I have this branch in my worktree (still a draft):
  https://github.com/jerome-pouiller/zephyr/tree/factorize-wpk-support

The idea is to factorize DT definitions common to all the radio boards. 3014cf7900e and 857d9c903f7 introduce regulators.

Reply:
How does this look now?